sdk/iOS/src/MSClient.m

```diff
@@ -67,90 +67,82 @@ -(NSString *) installId
-                                     applicationKey:key];
-}
+#pragma mark * Public Initializer Methods
+
 
-+(MSClient *) clientWithApplicationURL:(NSURL *)applicationUrl
-                            gatewayURL:(NSURL *)gatewayUrl
-                        applicationKey:(NSString *)key
+-(id) initWithApplicationURL:(NSURL *)url
 {
-    return [[MSClient alloc] initWithApplicationURL:applicationUrl
-                                         gatewayURL:gatewayUrl
-                                     applicationKey:key];
+    self = [super init];
+    if(self)
+    {
+        _applicationURL = url;
+        _login = [[MSLogin alloc] initWithClient:self];
+        _push = [[MSPush alloc] initWithClient:self];
+    }
+    return self;
 }
 
-#pragma mark * Public Initializer Methods
+@synthesize loginHost = _loginHost;
+-(void) setLoginHost:(NSURL *)loginHost
+{
+    if (loginHost.path.length > 0) {
+        @throw [NSException exceptionWithName:@"InvalidLoginHost"
+                                       reason:[NSString stringWithFormat:@"Login host should not include a path portion"]
+                                     userInfo:nil];
+    }
 
+    if (![loginHost.scheme isEqualToString:@"https"]) {
+        @throw [NSException exceptionWithName:@"InvalidLoginHost"
+                                       reason:[NSString stringWithFormat:@"Login host must use the https scheme"]
+                                     userInfo:nil];
+    }
+    
+    _loginHost = loginHost;
+}
 
--(id) initWithApplicationURL:(NSURL *)url
+-(NSURL *)loginHost
 {
-    return [self initWithApplicationURL:url gatewayURL:nil applicationKey:nil];
+    if (_loginHost == nil) {
+        NSString *host = [self.applicationURL.host stringByAddingPercentEncodingWithAllowedCharacters:
+                                [NSCharacterSet URLHostAllowedCharacterSet]];
+        
+        return [NSURL URLWithString:[NSString stringWithFormat:@"https://%@", host]];
+        return [self.applicationURL baseURL];
+    }
+    
+    return _loginHost;
 }
 
--(id) initWithApplicationURL:(NSURL *)url applicationKey:(NSString *)key
+@synthesize loginPrefix = _loginPrefix;
+- (void) setLoginPrefix:(NSString *)loginPrefix
 {
-    return [self initWithApplicationURL:url gatewayURL:nil applicationKey:key];
+    _loginPrefix = [loginPrefix stringByAddingPercentEncodingWithAllowedCharacters:[NSCharacterSet URLPathAllowedCharacterSet]];
 }
 
--(id)initWithApplicationURL:(NSURL *)applicationUrl
-                 gatewayURL:(NSURL *)gatewayUrl
-             applicationKey:(NSString *)key
+-(NSString *) loginPrefix
 {
-    self = [super init];
-    if(self)
-    {
-        _applicationURL = applicationUrl;
-        _gatewayURL = gatewayUrl;
-        _applicationKey = [key copy];
-        _login = [[MSLogin alloc] initWithClient:self];
-        _push = [[MSPush alloc] initWithClient:self];
+    if (!_loginPrefix) {
+        _loginPrefix = @".auth/login";
```

Comment:
Yes, appendByPathComponent, will add a slash if neither the original url or the path components has a slash. If one of them has it, then it will not.